src/_pytest/python_api.py

```diff
@@ -651,12 +651,9 @@ def raises(expected_exception, *args, match=None, **kwargs):
     message = "DID NOT RAISE {}".format(expected_exception)
 
     if not args:
-        if kwargs:
-            msg = "Unexpected keyword arguments passed to pytest.raises: "
-            msg += ", ".join(sorted(kwargs))
-            msg += "\nUse context-manager form instead?"
-            raise TypeError(msg)
-        return RaisesContext(expected_exception, message, match)
+        return RaisesContext(
+            expected_exception, message=message, match_expr=match, **kwargs
```

Comment:
`RaisesContext` doesn't take any arguments besides the three. Is it the intention that when kwargs are provided, this will fail with a `TypeError`? If so, I think a code comment would be useful, because it looks like a bug and trips up mypy (once it's made to check the function).

Same for `WarningsChecker` below.

Reply:
For reference, fixed in 866904ab8.